pkg/diskmaker/controllers/lvset/device_age.go

```diff
@@ -0,0 +1,62 @@
+package lvset
+
+import (
+	"sync"
+	"time"
+)
+
+var (
+	// deviceMinAge is the minimum age for a device to be considered safe to claim
+	// otherwise, it could be a device that some other entity has attached and we have not claimed.
+	deviceMinAge = time.Minute
+)
+
+type ageMap struct {
+	ageMap      map[string]time.Time
+	mux         sync.Mutex
+	currentTime *func() time.Time
+}
+
+func newAgeMap() *ageMap {
+	f := func() time.Time {
+		return time.Now()
+	}
```

Comment:
I think using a function pointer to set time for test purposes looks kind of weird. Not right now but may be we should consider defining mock interface for time and then using it.